tests/src/Simple/ForegroundThreading/ForegroundThreading.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+class Program
+{
+    public const int Pass = 100;
+    public const int Fail = -1;
+
+    static int Main()
+    {
+
+        return ForegroundThreadsWaiter.Run();
+    }
+}
+
+class ForegroundThreadsWaiter
+{
+
+    public static int Run()
```

Comment:
We do have tests for this in the big test suite from the main repo. I do not think it is worth protecting this by a new smoke tests. We would like to keep the number of smoke tests to be small.

If you would like to protect this via smoketest, you should be able to add it as the last thing to the existing threading smoketest.

Reply:
Do you mean that one?
https://github.com/dotnet/runtime/blob/master/src/mono/mono/tests/main-returns-background-change.cs
I can't find how this being ran for CoreRT.

Reply:
ha, you are right. We do not seem to have a test for this.

I remembered that we have fixed a bug around this in CoreCLR (https://github.com/dotnet/coreclr/pull/2164) and I have incorrectly assumed that we must have a test for this too.

Could you please submit a PR with your test for this to http://github.com/dotnet/runtime ? It can go under https://github.com/dotnet/runtime/tree/master/src/tests/baseservices/threading/regressions

Reply:
No worries, will do.